feature/client-charge/src/commonMain/kotlin/org/mifos/mobile/feature/charge/charges/ClientChargeScreen.kt

```diff
@@ -206,11 +284,339 @@ private fun ClientChargeDialogs(
+        "Savings" -> state.savingsAccounts
+        "Loan" -> state.loanAccounts
+        "Shares" -> state.shareAccounts
+        else -> emptyList()
+    }
+
+    // 3. UI Layout
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = 24.dp, vertical = 16.dp),
+    ) {
+        FilterHeader(onClear = onClear)
+
+        HorizontalDivider(modifier = Modifier.padding(vertical = 8.dp))
+
+        if (state.canSwitchAccounts) {
+            AccountTypeSection(
+                selectedTabLabel = selectedTabLabel,
+                onTabSelected = { newTab ->
+                    selectedTabLabel = newTab
+                    selectedAccountObject = null
+                },
+            )
+
+            if (currentAccountList.isNotEmpty()) {
+                AccountDropdownSection(
+                    label = selectedTabLabel,
+                    accounts = currentAccountList,
+                    selectedAccount = selectedAccountObject,
+                    onAccountSelected = { selectedAccountObject = it },
+                )
+            }
+        }
+
+        ChargeStatusSection(
+            selectedFilter = selectedFilter,
+            onFilterSelected = { selectedFilter = it },
+        )
+
+        Spacer(modifier = Modifier.height(40.dp))
+
+        FilterApplyButton(
+            onClick = {
+                val targetType = when (selectedTabLabel) {
+                    "Savings" -> ChargeType.SAVINGS
+                    "Loan" -> ChargeType.LOAN
+                    "Shares" -> ChargeType.SHARE
+                    else -> ChargeType.CLIENT
+                }
+                onApply(selectedAccountObject, targetType, selectedFilter)
+            },
+        )
+        Spacer(modifier = Modifier.height(24.dp))
+    }
+}
+
+@Composable
+private fun FilterHeader(onClear: () -> Unit) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Text(
+            text = "Filter Charges",
+            style = MaterialTheme.typography.titleLarge.copy(
+                fontWeight = FontWeight.Bold,
+                fontSize = 20.sp,
+            ),
+        )
+        TextButton(onClick = onClear) {
+            Text(
+                text = "Clear All",
+                style = MaterialTheme.typography.bodyMedium.copy(
+                    color = MaterialTheme.colorScheme.primary,
+                    fontWeight = FontWeight.SemiBold,
+                ),
+            )
+        }
+    }
+}
+
+@Composable
+private fun AccountTypeSection(
+    selectedTabLabel: String,
+    onTabSelected: (String) -> Unit,
+) {
+    Column {
+        Text(
+            text = "Account Type:",
+            style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Medium),
+            modifier = Modifier.padding(vertical = 8.dp),
```

Comment:
Use `DesignToken.padding` or `KptTheme.spacing` here:

```suggestion
            modifier = Modifier.padding(vertical = KptTheme.spacing.sm),
```

For padding and spacing, one can use `KptTheme.spacing` for the values present in it and for other values not available in `KptTheme` you can use `DesignToken.padding` for padding and `DesignToken.spacing` for spacing.

You can refer to this comment for detailed mapping between `DesignToken` and `KptTheme` for the right parameter choice:
https://github.com/openMF/mifos-mobile/pull/3013#issuecomment-3675564107

Here are the files:
1. DesignToken.kt: https://github.com/openMF/mifos-mobile/blob/97c3695885c466db33982408fdaea241d2cd4015/core/designsystem/src/commonMain/kotlin/org/mifos/mobile/core/designsystem/theme/DesignToken.kt

2. KptColorSchemeImpl.kt: https://github.com/openMF/mifos-mobile/blob/development/core-base/designsystem/src/commonMain/kotlin/template/core/base/designsystem/theme/KptColorSchemeImpl.kt

Please refer to existing feature modules.